src/components/structures/RoomView.tsx

```diff
@@ -345,22 +350,26 @@ function LocalRoomCreateLoader(props: ILocalRoomCreateLoaderProps): ReactElement
     return (
         <div className="mx_RoomView mx_RoomView--local">
             <ErrorBoundary>
-                <RoomHeader
-                    room={context.room}
-                    searchInfo={undefined}
-                    inRoom={true}
-                    onSearchClick={null}
-                    onInviteClick={null}
-                    onForgetClick={null}
-                    e2eStatus={props.localRoom.encrypted ? E2EStatus.Normal : undefined}
-                    onAppsClick={null}
-                    appsShown={false}
-                    excludedRightPanelPhaseButtons={[]}
-                    showButtons={false}
-                    enableRoomOptionsMenu={false}
-                    viewingCall={false}
-                    activeCall={null}
-                />
+                {SettingsStore.getValue("feature_new_room_decoration_ui") ? (
```

Comment:
Could we monitor/watch the setting instead of requiring a refresh?

Reply:
I'd rather skip this change altogether, i do not want to add more logic that i need to write tests for.